server/filestore.go

```diff
@@ -3415,6 +3428,7 @@ func (fs *fileStore) NumPending(sseq uint64, filter string, lastPerSubject bool)
 		if shouldExpire {
 			mb.tryForceExpireCacheLocked()
 		}
+		mb.finishedWithCache()
```

Comment:
How does this interact with tryForceExpire above? Feels like this should take that into consideration?

Reply:
If `mb.tryForceExpireCacheLocked()` successfully cleaned up `mb.cache` then `mb.finishedWithCache()` is a no-op.